### ISSUE TITLE
Add hierarchical/tree view of SOI

### ISSUE THREAD

**Issue by rsthornton** [halcyonic-systems/bert-old#11]
Develop a separate tab/window where the user can view the SOI as a complete hierarchy tree, as depicted on textbook page 476.

![Screenshot 2024-11-16 at 9 27 01 AM](https://github.com/user-attachments/assets/4691edab-1122-4655-accf-c6edbe28d011)

**Comment by Jtensminger** [comment]
@rsthornton The left-hand side of the diagram that shows the system-subsystem hierarchy seems straight forward. Is it necessary to include the right-hand side that shows the zoomed in flows and relations?

**Comment by rsthornton** [comment]
For the first iteration, I say no it definitely isn't necessary. The simple left-hand side version is fine for now. 

But I would like to tag this as a research issue and explore any potential benefits for adding zoom-in capability later on.

**Comment by rsthornton** [comment]
> @rsthornton The left-hand side of the diagram that shows the system-subsystem hierarchy seems straight forward. Is it necessary to include the right-hand side that shows the zoomed in flows and relations?

For the first iteration, I say no it definitely isn't necessary. The simple left-hand side version is fine for now.

But I would like to tag this as a research issue and explore any potential benefits to adding zoom-in capability later on.

**Comment by Jtensminger** [comment]
@rsthornton According to the diagram shown, it only shows sources and sinks at level -1 (top level). From reading the book, there doesn't appear to be any guidance on showing internal sources and sinks that haven't been formed into subsystems yet (see screenshot below)

![image](https://github.com/user-attachments/assets/c32df0df-115e-4353-af90-7c8d7cf7216d)

We could ignore them / exclude them from the diagram. What do you think?

**Comment by rsthornton** [comment]
@Jtensminger Yes, let's ignore them for now. Are there any other simple ways we could allow users to view associated sinks/sources when viewing the tree diagram? I don't feel attached to any approach, but think it could be useful.

**Comment by Jtensminger** [comment]
@rsthornton We can proceed under the assumption that we can ignore them for now, but I'll surface any ideas that come up during development.

**Comment by rsthornton** [comment]
Addressed in latest release.
